app/components/SendModal/index.js

```diff
@@ -54,25 +64,75 @@ class SendModal extends Component {
       transactionHash: '',
       isSending: false,
       to: '',
+      hip2Input: false,
+      hip2To: '',
+      hip2Error: '',
       amount: '',
       errorMessage: '',
       addressError: false,
       feeAmount: 0,
       txSize: 0,
     };
+    console.log(`HIP-2 Resolver: 127.0.0.1:${networkPorts[props.network]}`)
+    hip2.setServers([`127.0.0.1:${networkPorts[props.network]}`])
   }
 
   componentDidMount() {
     analytics.screenView('Send');
   }
 
-  updateToAddress = e => {
-    this.setState({to: e.target.value, errorMessage: ''});
-    if (e.target.value.length > 2 && !isValidAddress(e.target.value, this.props.network)) {
-      this.setState({errorMessage: this.context.t('invalidAddressPrefix')});
+  openLinkHandler (e) {
+    e.preventDefault()
+    shell.openExternal(e.target.href)
+  }
+
+  updateToAddress = async e => {
+    const dnsEnabled = !this.props.noDns
```

Comment:
Didn't really occur to me that DNS Server must be enabled in settings for HIP-2 resolution to work until I read the code. Is there any way we can tell users this in ui?

Reply:
Also, currently, DNS servers aren't enabled by default. Should we change `noDns` default from `yes` to `no`? i.e. enable dns by default.